src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/MasterServiceMetrics.java

```diff
@@ -58,12 +63,37 @@ public void collectMetrics(long startTime) {
                 return;
             }
 
+            /*
+            pendingTasks API returns object of PendingClusterTask which contains insertOrder, priority, source, timeInQueue.
+                Example :
+                     insertOrder: 101,
+                     priority: "URGENT",
+                     source: "create-index [foo_9], cause [api]",
+                     timeIn_queue: "86ms"
+             */
+
+            List<PendingClusterTask> pendingTasks = ESResources.INSTANCE.getClusterService().getMasterService()
+                    .pendingTasks();
+            HashMap<String,Integer> pendingTaskCountPerTaskType = new HashMap<>();
+
+            pendingTasks.stream().forEach( pendingTask -> {
+                String pendingTaskType = pendingTask.getSource().toString().split(" ",2)[0];
+                if (pendingTaskCountPerTaskType.containsKey(pendingTaskType)) {
+                    pendingTaskCountPerTaskType.put(pendingTaskType, pendingTaskCountPerTaskType.get(pendingTaskType) + 1);
+                }
+                else{
+                    pendingTaskCountPerTaskType.put(pendingTaskType,1);
+                }
```

Comment:
Can we use `map.put(key, map.getOrDefault(key, 0) + 1);` instead?

Reply:
Done